### PR TITLE
CLDR-17492 Update CLDR to not use utilities-for-cldr anymore

### PR DESCRIPTION
**Pull request by mihnita**
CLDR-17492

WARNING: all the code changes are OK, but I need some help / opinions.

---

1. According to the plan, `UOption` and `ElapsedTimer` are not used by ICU anymore, and are moved to CLDR.

I've put them in `tools/cldr-code/src/main/java/com/ibm/icu/dev/util/`
If that does not seem like the right place, let me know and I will move them.

---

2. `UOption` is used many times in `tools/cldr-code`, but also used once in `tools/cldr-apps`

More precisely by `tools/cldr-apps/src/test/java/org/unicode/cldr/unittest/web/perf/PerfTest.java`

That was a fork of the same class in ICU, in [`icu4j/perf-tests/src/main/java/com/ibm/icu/dev/test/perf/PerfTest.java`](https://github.com/unicode-org/icu/blob/main/icu4j/perf-tests/src/main/java/com/ibm/icu/dev/test/perf/PerfTest.java)

But it looks like it was forked, and never used, except by `tools/cldr-apps/src/test/java/org/unicode/cldr/unittest/web/perf/NullPerf.java`, which looked like a placeholder, doing nothing.

One option was to put `UOption` in some common place, where it can be used by both `tools/cldr-apps` and `tools/cldr-code`

The other option was to remove that unused code from `tools/cldr-apps`

I went with the deletion.

---

3. The tests fail because of new (Unicode 16) scripts

Between the previous ICU 76.0.1 snapshot and now the ICU repo was updated with some Unicode 16 data, which also adds a few scripts.

See [`com.ibm.icu.lang.UScript` here](https://github.com/unicode-org/icu/commit/423b4c7c313cbbd085d41c207dd8ac53f22956ab#diff-ab48615a0929d4d2661e36a89704e70a1fe3fafc5136b4f65f38e054346a37c2)

I added (temporarily) the missing entries to `common/supplemental/likelySubtags.xml` (in this PR),
between special comments:

```xml
<!-- START: Added by Unicode 16, temporary -->
...
 <!-- END: Added by Unicode 16, temporary -->
```

I don't know what the right process is to add such scripts, I only did it that the code changes are fine and the PR doesn't fail.
Let me know if you have a better way to do this, and I'll do it.

It just looked like a chicken-and-egg problem.
Without this PR CLDR can't start using the new ICU (and stop using utilities-for-cldr), but this CL can't go in with the old ICU (because `UnicodeMap` is now in ICU proper)
So that was my way to break the impasse, temporarily add them.

---

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: common/supplemental/likelySubtags.xml
Comment (srl295) [inline common/supplemental/likelySubtags.xml:2727]: @mihnita this is generated, so it'll come out in the wash later

File: tools/cldr-apps/pom.xml
Comment (srl295) [inline tools/cldr-apps/pom.xml:71]: yay!

**Review by srl295 — APPROVED** [review]

File: tools/cldr-code/src/main/java/com/ibm/icu/dev/util/ElapsedTimer.java
Comment (srl295) [inline tools/cldr-code/src/main/java/com/ibm/icu/dev/util/ElapsedTimer.java:6]: found the source of trouble..

File: tools/cldr-code/src/main/java/com/ibm/icu/dev/util/UOption.java
Comment (srl295) [inline tools/cldr-code/src/main/java/com/ibm/icu/dev/util/UOption.java:13]: I think it's fine to have these here. We can rename them etc later as needed.

**Review by srl295 — COMMENTED** [review]
@mihnita something really critical is - does unicodetools still build with this? It needs to build or needs to build soon.

**Comment by srl295** [comment]
@mihnita all of your choices LGTM as implemented.  like to ensure unicodetools is going to build before merging.

File: common/supplemental/likelySubtags.xml
Comment (mihnita) [reply common/supplemental/likelySubtags.xml:2727]: Ack.

I suspect the generation will bring these scripts back (and do it with the correct comment)
This is here only to make the tests pass.

**Comment by mihnita** [comment]
> like to ensure unicodetools is going to build before merging

I have a similar PR for unicodetools:
https://github.com/unicode-org/unicodetools/pull/810

Has one failure, and it does not look like it is from this refactoring.
Unicodetools used ICU 72.0.1-SNAPSHOT-cldr-2022-08-17, and I am trying to jump directly to 76.0.1-SNAPSHOT

Working with Markus to see why it fails.

Feel free to add yourself to that PR, if you want.
I don't have rights to add people there.

Thanks,
M.

**Review by srl295 — APPROVED** [review]

**Comment by srl295** [comment]
@mihnita do you have  perms to merge this (if so go ahead) otherwise I will  if it's ready

**Comment by mihnita** [comment]
> @mihnita do you have perms to merge this (if so go ahead) otherwise I will if it's ready

I don't, I just wanted to send a message.

Thank you.

File: tools/cldr-code/src/main/java/com/ibm/icu/dev/util/ElapsedTimer.java
Comment (markusicu) [inline tools/cldr-code/src/main/java/com/ibm/icu/dev/util/ElapsedTimer.java:7]: Will CLDR now also have to carry the pre-2016 IBM ICU license?
Maybe it does already?

**Review by markusicu — APPROVED** [review]

**Comment by srl295** [comment]
@mihnita sub-PR https://github.com/mihnita/cldr/pull/1 - rewrite ElapsedTimer

File: tools/cldr-code/src/main/java/com/ibm/icu/dev/util/ElapsedTimer.java
Comment (srl295) [reply tools/cldr-code/src/main/java/com/ibm/icu/dev/util/ElapsedTimer.java]: Yanked

**Comment by mihnita** [comment]
Should I revert the new `ElapsedTimer`, land this PR, and refactor it later?
Or is there a license problem?

**Comment by srl295** [comment]
> Should I revert the new `ElapsedTimer`, land this PR, and refactor it later? Or is there a license problem?

Yes I'm trying to avoid a license problem.

**Comment by srl295** [comment]
> > Should I revert the new `ElapsedTimer`, land this PR, and refactor it later? Or is there a license problem?
> 
> Yes I'm trying to avoid a license problem.

try again https://github.com/mihnita/cldr/pull/2

**Comment by mihnita** [comment]
All tests pass with the new one, thanks.

Do you want me to get this to local, squash, and `push -f`?
(to eliminate the `Bad commit message: “Merge pull request #2 from ...`)

Or you can do it from GitHub (I don't have the rights to do it)

From my side it is ready to merge.

Thank you,
M.

**Review by markusicu — APPROVED** [review]

**Review by srl295 — APPROVED** [review]

**Comment by srl295** [comment]
@mihnita LGTM.  Let's wait until after shakedown (starting tomorrow) to merge.

**Comment by macchiati** [comment]
We are launching Shakedown tomorrow morning. So we should not merge this, just in case it could cause instabilities. Friday should be fine. I'm converting to draft just so someone doesn't inadvertently merge.

**Comment by mihnita** [comment]
Thank you everybody!
Mihai

**Pull request merged**
